scala3doc-js/src/searchbar/SearchbarComponent.scala

```diff
@@ -0,0 +1,86 @@
+package dotty.dokka
+
+import org.scalajs.dom._
+import org.scalajs.dom.html.Input
+
+class SearchbarComponent(val callback: (String) => List[PageEntry]):
+  extension (p: PageEntry)
+    def toHTML =
+      val wrapper = document.createElement("div")
+      wrapper.classList.add("scala3doc-searchbar-result")
+      wrapper.classList.add("monospace")
+
+      val resultA = document.createElement("a").asInstanceOf[html.Anchor]
+      resultA.href = Globals.pathToRoot + p.location
+      resultA.text = s"${p.name}"
+
+      val location = document.createElement("span")
+      location.classList.add("pull-right")
+      location.classList.add("scala3doc-searchbar-location")
+      location.textContent = p.description
+
+      wrapper.appendChild(resultA)
+      wrapper.appendChild(location)
+      wrapper
+
+  def handleNewQuery(query: String) =
+    val result = callback(query).map(_.toHTML)
+    while (resultsDiv.hasChildNodes()) resultsDiv.removeChild(resultsDiv.lastChild)
+    result.foreach(resultsDiv.appendChild)
```

Comment:
This would probably be faster with a document fragment (created by `document.createDocumentFragment()`). You could append all the results to the fragment and then do `resultsDiv.appendChild(fragment)` to add all the results to the div at once.